ivadomed/utils.py

```diff
@@ -745,3 +809,95 @@ def save_feature_map(batch, layer_name, context, model, test_input, slice_axis):
         nib_pred = nib.Nifti1Image(attention_map, nib_ref.affine)
 
         nib.save(nib_pred, save_directory)
+
+
+def save_color_labels(gt_data, binarize, gt_filename, output_filename, slice_axis):
+    rdict = {}
+    n_class, h, d, w = gt_data.shape
+    labels = range(n_class)
+    # Generate color labels
+    multi_labeled_pred = np.zeros((h, d, w, 3))
+    if binarize:
+        rdict['gt'] = threshold_predictions(gt_data)
+
+    # Keep always the same color labels
+    np.random.seed(6)
+
+    for label in labels:
+        r, g, b = np.random.randint(0, 256, size=3)
+        multi_labeled_pred[..., 0] += r * gt_data[label, ]
+        multi_labeled_pred[..., 1] += g * gt_data[label, ]
+        multi_labeled_pred[..., 2] += b * gt_data[label, ]
+
+    rgb_dtype = np.dtype([('R', 'u1'), ('G', 'u1'), ('B', 'u1')])
+    multi_labeled_pred = multi_labeled_pred.copy().astype('u1').view(dtype=rgb_dtype).reshape((h, d, w))
+
+    save_nii(multi_labeled_pred.transpose((1, 2, 0)), [], gt_filename,
+             output_filename, slice_axis=slice_axis, unet_3D=True, binarize=False)
+
+    return multi_labeled_pred
+
+
+def pil_list_to_numpy(pil_list):
+    n_classes = len(pil_list)
+    w, h = pil_list[0].size
+    numpy_array = np.zeros((w, h, n_classes))
+    for idx, pil_img in enumerate(pil_list):
+        numpy_array[..., idx] = np.array(pil_img)
+    return numpy_array
+
+
+def convert_labels_to_RGB(grid_img):
+    # Keep always the same color labels
```

Comment:
![image](https://user-images.githubusercontent.com/49137243/77609120-8378db00-6ef5-11ea-86ff-5f0677c5c237.png)
This is necessary to have a decent view if the number of classes is not 1 or 3.